woocommerce/woocommerce-ios
# Store creation: Update profiler flow

Closes: #10390 


## Description

This PR updates the new flow for store creation to start the creation at the beginning and show profiler questions afterward. Since the order of questions have changed compared to the old flow, and the current code is hard to maintain, I added a separate container view for the profiler flow. The new flow will start this coordinator after completing the request to create a store and enable the free trial plan.

Also: the code has been cleaned up to use the `@MainActor` attribute only when necessary.

## Testing instructions

- Enable the feature flag `optimizeProfilerQuestions`.
- Log in to the app and open the store picker.
- Scroll to bottom and select Add a store > Create a new store.
- Notice that the free trial summary view is displayed immediately.
- Proceed to start the free trial plan. After this is complete, you should be navigated to the profiler flow.
- Notice that the first question is selling status, followed by store category and country code. The new questions are displayed at the end.
- The progress view should be displayed when all questions have been responded to. The app should then navigate to the dashboard screen when the store is ready.

Edge case: stay on one of the profiler questions until the store is ready (around 1-2 minutes). When completing the profiler flow, you should be navigated immediately to the dashboard screen.

Please feel free to test again with the feature flag off, the old flow should still work properly.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/86d63ab5-6e47-495a-81fb-fa4fb3c9528b





---
